python/mvdtool.cpp

```diff
@@ -155,31 +168,43 @@ PYBIND11_MODULE(mvdtool, mvd) {
              })
         .def("positions", [](const File& f) {
                 auto res = f.getPositions(Range::all());
-                return py::array({res.shape()[0], res.shape()[1]}, res.data());
+                return py::array({res.shape()[0], POSITION_WIDTH}, res.data());
              })
         .def("positions", [](const File& f, int offset) {
                 Range r(offset, 1);
                 auto res = f.getPositions(r);
-                return py::array({res.shape()[0], res.shape()[1]}, res.data());
+                return py::array({res.shape()[0], POSITION_WIDTH}, res.data());
              })
         .def("positions", [](const File& f, int offset, int count) {
                 Range r(offset, count);
                 auto res = f.getPositions(r);
-                return py::array({res.shape()[0], res.shape()[1]}, res.data());
+                return py::array({res.shape()[0], POSITION_WIDTH}, res.data());
+             })
+        .def("positions", [](const File& f, const pyarray<size_t>& idx) {
+                using position_t = std::array<typename Positions::element, POSITION_WIDTH>;
+                const auto& func = [&f](const MVD::Range& r){return f.getPositions(r);};
+                const auto res = _atIndices<position_t>(func, f.size(), idx);
+                return py::array({res.size(), POSITION_WIDTH}, res.empty()? nullptr: res[0].data());
```

Comment:
Nitpicking here but `front` does not assume `res` is a random access container.
```suggestion
                return py::array({res.size(), POSITION_WIDTH}, res.empty()? nullptr: res.front().data());
```